mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ModClusterConfigurationServiceBuilder.java

```diff
@@ -206,14 +204,19 @@ public ServiceName getServiceName() {
                 .setTtl(TTL.resolveModelAttribute(context, model).asInt())
                 .setNodeTimeout(NODE_TIMEOUT.resolveModelAttribute(context, model).asInt())
         ;
-        optionalString(BALANCER.resolveModelAttribute(context, model)).ifPresent(balancer -> builder.node().setBalancer(balancer));
-        optionalString(LOAD_BALANCING_GROUP.resolveModelAttribute(context, model)).ifPresent(group -> builder.node().setLoadBalancingGroup(group));
-
-        optionalList(PROXIES.resolveModelAttribute(context, model)).ifPresent(
-                refs -> refs.stream()
-                        .map(ModelNode::asString)
-                        .forEach(ref -> this.outboundSocketBindings.add(new InjectedValueDependency<>(CommonUnaryRequirement.OUTBOUND_SOCKET_BINDING.getServiceName(context, ref), OutboundSocketBinding.class)))
-        );
+        ModelNode node = BALANCER.resolveModelAttribute(context, model);
+        if (node.isDefined()) {
+            builder.node().setBalancer(node.asString());
+        }
+        node = LOAD_BALANCING_GROUP.resolveModelAttribute(context, model);
+        if (node.isDefined()) {
+            builder.node().setLoadBalancingGroup(node.asString());
+        }
+        node = PROXIES.resolveModelAttribute(context, model);
+        if (node.isDefined()) for (ModelNode ref : node.asList()) {
```

Comment:
Please fix for readability.

Reply:
What specifically is objectionable?

Reply:
Ah, I see.  That's idiomatic to me, but I'll update it.